python_modules/dagster/dagster/_core/definitions/declarative_automation/automation_condition_evaluator.py

```diff
@@ -132,7 +136,16 @@ def evaluate(self) -> Tuple[Sequence[AutomationResult], Sequence[EntitySubset[En
                 f"requested ({requested_str}) "
                 f"({format(result.end_timestamp - result.start_timestamp, '.3f')} seconds)"
             )
-            num_evaluated += 1
+
+        for topo_level in self.asset_graph.toposorted_entity_keys_by_level:
+            coroutines = [
+                _evaluate_entity_async(entity_key, num_evaluated)
```

Comment:
offset should vary by index in the list (i.e. just enumerate the entity keys in the topo level)